### ISSUE TITLE
Odd crash in Android demo

### ISSUE THREAD

**Issue by kuza55** [tensorflow/tensorflow#2597]
### Environment info
Android Demo from 57658eddfab7763058c5316b1af41e77b1bb3b17

I've been trying to build an image classifier for android, and I ran into this exception in the TensorflowClassifier class.

I haven't run into this before despite playing with this for a few days, but I added some logging and will upload more details if I can repro it.

Not sure if anyone cares, but I thought I might as well capture this.

### Logs or other output that would be helpful
Caused by: java.lang.NumberFormatException: Invalid float: "Android"
   at java.lang.StringToReal.invalidReal(StringToReal.java:63)
   at java.lang.StringToReal.initialParse(StringToReal.java:164)
   at java.lang.StringToReal.parseFloat(StringToReal.java:323)
   at java.lang.Float.parseFloat(Float.java:306)
   at org.tensorflow.demo.TensorflowClassifier.recognizeImage(TensorflowClassifier.java:51)

FWIW the line number in recognizeImage is wrong since I've made changes to the file, but shouldn't have caused this.

[EDIT]: I found that the log message I was getting was "Android system is not using RGBA_8888 in default", I guess I'll see what that's about.

**Comment by andrewharp** [comment]
Hi,
The native classifier code returns the results as a String, which is parsed by TensorflowClassifier.java. It sounds like something has changed about the return format with your new model. Can you post the entire result string?

I don't think the RGBA_8888 warning is related. The bitmaps are created in 8888 format explicitly, so they don't rely on the system default.

**Comment by kuza55** [comment]
Sorry for the confusion, I have been modifying the android demo to recognize the photos stored on my device rather than from the camera and it seems that when the classifyImageBmp function in tensorflow_jni.cc gets an image that is not RGBA_8888, it returns the string "Android system is not using RGBA_8888 in default", which causes the exception above.

It might be good to make the JNI code raise an exception, rather than return a string that gets badly parsed, but otherwise this seems fine (though there is the TODO(jiayq): deal with other formats if necessary.)

**Comment by andrewharp** [comment]
Thanks for the clarification, makes sense now.

It should be pretty easy to convert an rgb565 (or whatever) bitmap to rgba8888 prior to sending it to the classifier.

Your suggestion to raise an exception is a good idea, I'll see about adding that.

**Comment by andrewharp** [comment]
Native error conditions have been changed to LOG(FATAL) messages  in https://github.com/tensorflow/tensorflow/commit/c32ef5a6b7bbcacfe39f20bc7bea9514a505c74f, so that errors can be caught as soon as they happen now.

I decided against adding 565 support for now because it's a little bit application specific how the resulting values are handled. Probably best for now just to transform to RGBA8888 before passing to the native code, or making custom changes to tensorflow-jni.cc to handle the bitmap appropriately.
